network.go

```diff
@@ -53,73 +53,66 @@ func linkByHwAddr(netHandle *netlink.Handle, hwAddr string) (netlink.Link, error
 	return nil, fmt.Errorf("Could not find the link corresponding to HwAddr %q", hwAddr)
 }
 
-func updateInterfaceAddrs(netHandle *netlink.Handle, link netlink.Link, addrs []*pb.IPAddress, add bool) error {
-	for _, addr := range addrs {
-		netlinkAddrStr := fmt.Sprintf("%s/%s", addr.Address, addr.Mask)
+func updateLink(netHandle *netlink.Handle, link netlink.Link, iface *pb.Interface) error {
+
+	// As a first step, clear out any existing addresses associated with the link:
```

Comment:
With clear and re-add we will lose any routes related to those addresses(e.g. default route via a gateway address in the same subnet). IMHO, we should compute diff of addresses and apply it instead.

Reply:
Yes, this could happen

Reply:
Good catch on the default route -- let me test this and update accordingly.

Reply:
@liwei ; @WeiZhang555 I did some testing locally, and I'm not sure if this should be a concern here.  

The default route will get created still for the subset when adding the address.  I verified this in network_test.go.  Initially we have a single address, and then in later 'test' on this same interface we add two more.    When doing so, you'll see that the default route created still exists for the initial address, as well as for the newly created addresses.

The non-default routes need to be added once the addresses have been populated.  If you are changing an interfaces IP information after this point, then the final state of routes for that interface should be sent again as well (I don't see this as a normal use case...).  

I don't see how we'd lose routes.